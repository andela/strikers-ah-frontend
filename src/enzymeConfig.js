import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
