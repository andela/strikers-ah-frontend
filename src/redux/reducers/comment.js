import { ADD_COMMENT, GET_COMMENTS } from '../actionTypes/commentTypes';

const initialState = {};
export default (state = initialState, { type, payload }) => {
  switch (type) {
    case ADD_COMMENT:
      return {
        ...state,
        ...payload,
      };
    case GET_COMMENTS:
      return {
        ...state,
        ...payload,
      };
    default:
      return state;
  }
};
