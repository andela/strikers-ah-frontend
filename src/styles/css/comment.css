*,
*::before,
*::after {
  box-sizing: border-box;
}

body {
  margin: 0;
  padding: 0;
}

.comment-form {
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  border-radius: 10px;
  border: none;
}
.comment-form textarea {
  width: 100%;
  resize: none;
  min-height: 6em;
  border: none;
  outline: none;
  font-size: 1.2em;
  color: #444;
  box-shadow: 0px 0px 14px #f5f3f3;
  border-radius: 5px;
  padding: 1%;
}
.comment-form .form-button {
  font-size: 1.2em;
  line-height: 2em;
  width: 6em;
  border-radius: 10px;
  border: none;
  outline: none;
  background: #0396eb;
  color: white;
  transition: 0.5s;
  margin-top: 1%;
}
.comment-form .form-button:hover {
  transform: translateY(-2px);
  background: #15a1f1;
}
.comment-form .text-area-label {
  color: #292828;
  font-weight: bold;
  line-height: 40px;
}
