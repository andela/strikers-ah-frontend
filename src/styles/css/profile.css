@import url('https://fonts.googleapis.com/css?family=Open+Sans&display=swap');
*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Roboto Condensed', sans-serif;
}

body {
  background: #f0f0f0;
  font-family: 'Open Sans', sans-serif;
}

@keyframes showAddImage {
  from {
    background: transparent;
    opacity: 0;
  }
  to {
    background: #0788f1;
    opacity: 0.7;
  }
}

@keyframes modal {
  from {
    margin-top: 0;
    border-radius: 0;
    box-shadow: none;
  }
  to {
    margin-top: 3%;
    border-radius: 10px;
    box-shadow: 0px 0px 14px #333;
  }
}

.web-heading {
  color: #079bfd;
}

.profile-grid-off,
.profile-grid {
  padding: 2%;
  position: relative;
}
.profile-grid-off .user-account-info,
.profile-grid .user-account-info {
  padding: 1.5%;
  box-shadow: 0px 0px 14px rgba(191, 191, 191, 0.24);
  border-radius: 10px;
  background: white;
}
.profile-grid-off .user-account-info .statistics,
.profile-grid .user-account-info .statistics {
  width: 100%;
  border-top: 2px solid #ddd;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  text-align: center;
  margin-top: 1%;
  padding-top: 1%;
}
.profile-grid-off .user-account-info .statistics .stat-label,
.profile-grid .user-account-info .statistics .stat-label {
  text-decoration: none;
  color: #949292;
  display: inline-block;
}
.profile-grid-off .user-account-info .statistics .stat-label .value,
.profile-grid .user-account-info .statistics .stat-label .value {
  color: #333;
  font-weight: bold;
  font-size: 1.2em;
}
.profile-grid-off .user-account-info .user-account-flex,
.profile-grid .user-account-info .user-account-flex {
  position: relative;
  display: grid;
  grid-template-columns: 10em 1fr;
  grid-column-gap: 10px;
}
.profile-grid-off .user-account-info .user-account-flex .img-container,
.profile-grid .user-account-info .user-account-flex .img-container {
  position: relative;
  height: 10em;
  width: 10em;
  margin-right: 2em;
  text-align: center;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .img-container
  .add-img-toggler,
.profile-grid
  .user-account-info
  .user-account-flex
  .img-container
  .add-img-toggler {
  position: absolute;
  background: #ddd;
  bottom: 0;
  border: none;
  outline: none;
  left: 0;
  width: 100%;
  background: transparent;
  font-weight: bolder;
  color: transparent;
  font-size: 2em;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .img-container
  .add-img-toggler:hover,
.profile-grid
  .user-account-info
  .user-account-flex
  .img-container
  .add-img-toggler:hover {
  animation-name: showAddImage;
  animation-duration: 0.5s;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  color: white;
}
.profile-grid-off .user-account-info .user-account-flex .img-container img,
.profile-grid .user-account-info .user-account-flex .img-container img {
  width: 100%;
  height: 100%;
  cursor: pointer;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .img-container
  img:hover
  ~ .add-img-toggler,
.profile-grid
  .user-account-info
  .user-account-flex
  .img-container
  img:hover
  ~ .add-img-toggler {
  z-index: 1;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .img-container:hover
  .profile-image-label,
.profile-grid
  .user-account-info
  .user-account-flex
  .img-container:hover
  .profile-image-label {
  display: inline-block;
}
.profile-grid-off .user-account-info .user-account-flex .info-container,
.profile-grid .user-account-info .user-account-flex .info-container {
  padding: 1em 0;
  grid-area: 'info-container';
}
.profile-grid-off .user-account-info .user-account-flex .info-container .name,
.profile-grid .user-account-info .user-account-flex .info-container .name {
  font-weight: 500;
  font-size: 2em;
  text-transform: capitalize;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .info-container
  .username,
.profile-grid .user-account-info .user-account-flex .info-container .username {
  line-height: 40px;
  color: #949292;
}
.profile-grid-off .user-account-info .user-account-flex .info-container .bio,
.profile-grid .user-account-info .user-account-flex .info-container .bio {
  padding: 1% 10px 1% 0px;
  line-height: 30px;
  color: #535353;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .info-container
  .btn-follow,
.profile-grid
  .user-account-info
  .user-account-flex
  .info-container
  .btn-follow {
  float: right;
  margin-right: 10px;
  border: 1px solid #079bfd;
  width: 100px;
  padding: 5px;
  border-radius: 5px;
  color: #079bfd;
  margin-top: 10px;
  outline: none;
  font-size: 0.9em;
  transition: 0.2s;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .info-container
  .btn-follow:hover,
.profile-grid
  .user-account-info
  .user-account-flex
  .info-container
  .btn-follow:hover {
  box-shadow: 0px 0px 14px rgba(191, 191, 191, 0.24);
  background: #079bfd;
  color: white;
  transform: translateY(-2px);
}
.profile-grid-off .upload-image-modal,
.profile-grid .upload-image-modal {
  position: absolute;
  width: 100%;
  height: 70vh;
  top: 0;
  left: 50%;
  background: rgba(86, 86, 87, 0.7);
  transform: translateX(-50%);
  z-index: 100;
}
.profile-grid-off .upload-image-modal .upload-page,
.profile-grid .upload-image-modal .upload-page {
  position: relative;
  width: 90%;
  margin-top: 3%;
  margin-left: auto;
  margin-right: auto;
  background: #eff2f5;
  min-height: 90%;
  box-shadow: 0px 0px 14px #333;
  border-radius: 10px;
  animation-name: modal;
  animation-duration: 0.3s;
}
.profile-grid-off .upload-image-modal .btn,
.profile-grid .upload-image-modal .btn {
  font-size: 18px;
  line-height: 18px;
  height: 30px;
  padding: 0 10px;
  border-radius: 1px;
  margin: 1%;
}
.profile-grid-off .upload-image-modal .btn-cancel,
.profile-grid .upload-image-modal .btn-cancel {
  margin-left: 10%;
  border: 1px solid brown;
  color: #f38b8b;
}
.profile-grid-off .upload-image-modal .btn-cancel:hover,
.profile-grid .upload-image-modal .btn-cancel:hover {
  background: brown;
  color: white;
}
.profile-grid-off .upload-image-modal .btn-upload,
.profile-grid .upload-image-modal .btn-upload {
  background: white;
  border: 1px solid #079bfd;
  color: #079bfd;
}
.profile-grid-off .upload-image-modal .btn-upload:hover,
.profile-grid .upload-image-modal .btn-upload:hover {
  background: #079bfd;
  color: white;
}
.profile-grid-off .upload-image-modal .dropzone,
.profile-grid .upload-image-modal .dropzone {
  width: 100%;
  background: #e1f0ff;
  height: 100px;
  margin-bottom: 1%;
  position: relative;
}
.profile-grid-off .upload-image-modal .dropzone p,
.profile-grid .upload-image-modal .dropzone p {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
.profile-grid-off .upload-image-modal .crop .crop-section,
.profile-grid .upload-image-modal .crop .crop-section {
  width: 100%;
  display: grid;
  grid-template-columns: 60% 1% 30%;
  grid-column-gap: 1em;
}
.profile-grid-off .upload-image-modal .crop .cropper,
.profile-grid .upload-image-modal .crop .cropper {
  width: 100%;
  overflow: auto;
  height: 400px;
  text-align: center;
  border-radius: 10px;
}
.profile-grid-off .upload-image-modal .crop .separator,
.profile-grid .upload-image-modal .crop .separator {
  background: #dde4e9;
}
.profile-grid-off .upload-image-modal .crop .crop-preview,
.profile-grid .upload-image-modal .crop .crop-preview {
  width: 100%;
  overflow: hidden;
  text-align: center;
}
.profile-grid-off .upload-image-modal .crop .crop-preview canvas,
.profile-grid .upload-image-modal .crop .crop-preview canvas {
  border-radius: 10px;
  box-shadow: 0px 0px 14px #949292;
  border: 5px solid #ddd;
}
.profile-grid-off .left-profile,
.profile-grid .left-profile {
  background: white;
  box-shadow: 0px 0px 14px rgba(191, 191, 191, 0.24);
  border-radius: 10px;
}
.profile-grid-off .left-profile li,
.profile-grid .left-profile li {
  list-style-type: none;
  border-top: 1px solid #f0f0f0;
  line-height: 35px;
  padding-left: 10px;
}
.profile-grid-off .left-profile li .link,
.profile-grid .left-profile li .link {
  text-decoration: none;
  color: #949292;
}
.profile-grid-off .left-profile li .active,
.profile-grid .left-profile li .active {
  color: #079bfd;
}
.profile-grid-off .left-profile h3,
.profile-grid .left-profile h3 {
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  line-height: 40px;
  padding-left: 10px;
  background: rgba(9, 129, 241, 0.89);
  color: white;
}

.profile-grid {
  display: grid;
  grid-template-columns: 1fr 3fr;
  grid-column-gap: 2em;
  grid-template-areas: 'left-profile user-account-flex';
}

@media screen and (max-width: 1500px) {
  .profile-grid {
    width: 80%;
    margin: 2em auto;
    display: grid;
    grid-template-columns: 1fr 3fr;
    grid-column-gap: 10px;
    margin: 2em auto;
  }
}

@media screen and (max-width: 1000px) {
  .profile-grid {
    display: grid;
    grid-template-columns: 1fr 3fr;
    grid-column-gap: 4px;
    width: 98%;
    margin: 2em auto;
  }
}

@media screen and (max-width: 815px) {
  .profile-grid {
    width: 80%;
    display: grid;
    grid-template-columns: 1fr;
    grid-column-gap: 4px;
    width: 98%;
    margin: 2em auto;
  }
  .profile-grid .left-profile {
    display: none;
  }
}

@media screen and (max-width: 500px) {
  .profile-grid-off {
    display: grid;
    grid-template-columns: 1fr;
    width: 98%;
    margin: 2em auto;
  }
  .profile-grid-off .user-account-flex {
    display: grid;
    grid-template-columns: 1fr;
    grid-column-gap: 0px;
    grid-template-areas: 'image-container' 'info-container';
  }
  .profile-grid-off .user-account-flex .info-container {
    display: block;
    width: 100%;
  }
  .profile-grid-off .user-account-flex .info-container .name {
    font-weight: 100;
    font-size: 1em;
  }
  .profile-grid {
    display: block;
  }
}

.text-center {
  text-align: center;
}
