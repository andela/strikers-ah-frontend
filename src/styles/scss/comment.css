*,
*::before,
*::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}

body {
  margin: 0;
  padding: 0;
}

.comment-form {
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  border-radius: 10px;
  border: none;
  margin-top: 1%;
}

.comment-form textarea {
  width: 100%;
  resize: none;
  min-height: 4em;
  border: none;
  outline: none;
  font-size: 1.2em;
  color: #444;
  -webkit-box-shadow: 0px 0px 14px #f5f3f3;
          box-shadow: 0px 0px 14px #f5f3f3;
  border-radius: 5px;
  padding: 1%;
}

.comment-form .form-button {
  font-size: 1.2em;
  line-height: 2em;
  width: 6em;
  border-radius: 10px;
  border: none;
  outline: none;
  background: #0396eb;
  color: white;
  -webkit-transition: 0.5s;
  transition: 0.5s;
  margin-top: 1%;
}

.comment-form .form-button:hover {
  -webkit-transform: translateY(-2px);
          transform: translateY(-2px);
  background: #15a1f1;
}

.comment-form .text-area-label {
  color: #292828;
  font-weight: bold;
  line-height: 40px;
}

.comments-container {
  background: white;
  padding-bottom: 100px;
  margin-bottom: 10%;
}

.comment-container {
  position: relative;
  background: white;
  display: -ms-grid;
  display: grid;
  -ms-grid-columns: 4em 1fr;
      grid-template-columns: 4em 1fr;
  margin: 1em;
  border-radius: 10px;
  -webkit-box-shadow: 0px 0px 3px rgba(212, 220, 221, 0.658);
          box-shadow: 0px 0px 3px rgba(212, 220, 221, 0.658);
  padding-left: 1.2em;
  padding-top: 5px;
  width: 99%;
  margin-left: auto;
  margin-right: auto;
}

.comment-container .comment-form {
  position: absolute;
  width: 80%;
  background: #444;
  clear: both;
}

.comment-container .comment-form .input-container {
  background: green;
  position: relative;
}

.comment-container .comment-author {
  position: relative;
  height: 3em;
  width: 3em;
  border-radius: 10px;
  margin-right: 1em;
}

.comment-container .comment-author img {
  width: 100%;
  height: 100%;
  border-radius: 100%;
}

.comment-container .author-name {
  text-transform: capitalize;
  font-size: 1em;
  color: #43c0f1;
  font-weight: 520;
  line-height: 30px;
  cursor: pointer;
}

.comment-container .comment-body {
  width: 100%;
  padding: 0 1% 0 0;
}

.comment-container .comment-body p .time-stamp {
  color: darkgray;
}

.delete {
  color: #d1cccc;
  margin: 0 5px;
  cursor: pointer;
}

.delete:hover {
  color: red;
}

.edit {
  color: #b3afaf;
  margin: 0 5px;
  cursor: pointer;
}

.edit:hover {
  color: yellowgreen;
}

.color-success {
  color: #3f6408;
}

.edit-form-container {
  -webkit-transition: 0.5s;
  transition: 0.5s;
}
/*# sourceMappingURL=comment.css.map */