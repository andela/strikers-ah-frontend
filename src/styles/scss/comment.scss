*,
*::before,
*::after {
    box-sizing: border-box;
}

body {
    margin: 0;
    padding: 0;
}

.comment-form {
    width: 80%;
    margin-left: auto;
    margin-right: auto;
    border-radius: 10px;
    border: none;
    margin-top: 1%;

    textarea {
        width: 100%;
        resize: none;
        min-height: 4em;
        border: none;
        outline: none;
        font-size: 1.2em;
        color: #444;
        box-shadow: 0px 0px 14px rgb(245, 243, 243);
        border-radius: 5px;
        padding: 1%;
    }

    .form-button {
        font-size: 1.2em;
        line-height: 2em;
        width: 6em;
        border-radius: 10px;
        border: none;
        outline: none;
        background: rgb(3, 150, 235);
        color: white;
        transition: 0.5s;
        margin-top: 1%;

        &:hover {
            transform: translateY(-2px);
            background: rgb(21, 161, 241);
        }
    }

    .text-area-label {
        color: rgb(41, 40, 40);
        font-weight: bold;
        line-height: 40px;
    }
}

.comments-container {
    background: white;
    padding-bottom: 100px;
    margin-bottom: 10%;
}

.comment-container {
    background: white;
    padding-bottom: 100px;
    margin-bottom: 10%;
}

.comment-container {
    position: relative;
    background: white;
    display: grid;
    grid-template-columns: 4em 1fr;
    margin: 1em;
    border-radius: 10px;
    box-shadow: 0px 0px 3px rgba(212, 220, 221, 0.658);
    padding-left: 1.2em;
    padding-top: 5px;
    width: 99%;
    margin-left: auto;
    margin-right: auto;
    width: 100%;
}

.comment-form {
    position: absolute;
    width: 80%;
    background: #444;

    .input-container {
        background: green;
        position: relative;
    }

    clear: both;
}

.comment-author {
    position: relative;
    background: white;
    display: grid;
    grid-template-columns: 4em 1fr;
    margin: 1em;
    border-radius: 10px;
    box-shadow: 0px 0px 3px rgba(212, 220, 221, 0.658);
    padding-left: 1.2em;
    padding-top: 5px;
    width: 99%;
    margin-left: auto;
    margin-right: auto;

    .comment-form {
        position: absolute;
        width: 80%;
        background: #444;

        .input-container {
            background: green;
            position: relative;
        }

        clear: both;
    }

    .comment-author {
        position: relative;
        height: 3em;
        width: 3em;
        border-radius: 10px;
        margin-right: 1em;

        img {
            width: 100%;
            height: 100%;
            border-radius: 100%;
        }
    }

    .author-name {
        text-transform: capitalize;
        font-size: 1em;
        color: rgb(67, 192, 241);
        font-weight: 520;
        line-height: 30px;
        cursor: pointer;
    }

    .comment-body {
        width: 100%;
        padding: 0 1% 0 0;

        p {
            .time-stamp {
                color: darkgray;
            }
        }
    }
}

.delete {
    color: rgb(209, 204, 204);
    margin: 0 5px;
    cursor: pointer;

    &:hover {
        color: red;
    }
}

.edit {
    color: rgb(179, 175, 175);
    margin: 0 5px;
    cursor: pointer;

    &:hover {
        color: yellowgreen;
    }
}

.edit-form-container {
    transition: 0.5s;
}