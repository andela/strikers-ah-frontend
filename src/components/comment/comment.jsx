import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Moment from 'react-moment';
import 'moment-timezone';
/* eslint-disable react/destructuring-assignment */
import { connect } from 'react-redux';
import {
  faTrash,
  faPencilAlt,
  faTimes,
  faHistory,
  faEyeSlash,
  faThumbsUp,
} from '@fortawesome/free-solid-svg-icons';
import EditHistory from './editHistory';

import { getLoggedInUser } from '../../helpers/authentication';
import CommentForm from './commentForm';
import '../../styles/css/comment.css';
import LikeComment from '../../redux/actions/commentLike';

/**
 * @author Mwibutsa Floribert
 * @param {Integer} id  --
 * @returns { * } --
 */
export class Comment extends Component {
  /**
   *
   * @param {*} props
   */
  constructor(props) {
    super(props);
    this.likeComment = this.likeComment.bind(this);
  }

  likeComment = id => {
    const { slug } = this.props;
    this.props.LikeComment(slug, id);
  };

  handleEditComment = () => {
    const { toggleEditCommentForm } = this.props;
    toggleEditCommentForm(true);
  };

  /**
   * @author Mwibutsa Floribert
   * @param { object } object --
   * @param { string } role --
   * @returns { object } --
   */
  assignRole = (object, role) => (object.role ? object : { ...object, role });

  /**
   * @author Mwibutsa Floribert
   * @param { * } first --
   * @param {*} second --
   * @returns{ boolean } --
   */
  canEdit = (first, second) => first || second;

  /**
   * @author Mwibutsa Floribert
   * @returns { * } ---
   */
  render() {
    let user = getLoggedInUser();
    const {
      comment,
      deleteComment,
      slug,
      editComment,
      formId,
      toggleEditCommentForm,
      toggleEditHistory,
      editHistoryMode,
      commentHistoryId,
    } = this.props;
    const { author, comment: body, id, updatedAt, history, likes } = comment;

    if (user) {
      user = this.assignRole(user, 'User');
    }
    const { editMode } = this.props;
    if (editMode === true && formId === id) {
      return (
        <div className="edit-form-container">
          <FontAwesomeIcon
            icon={faTimes}
            className="delete"
            onClick={this.handleEditComment}
            test-data="closeButton"
          />
          {'   '}
          <span className="color-success">Editing</span>
          <CommentForm
            test-data="commentFormComponent"
            buttonLabel="save"
            slug={slug}
            currentValue={body}
            saveComment={(editedComment, articleSlug) => {
              toggleEditCommentForm(formId);
              return editComment(articleSlug, id, editedComment);
            }}
          />
        </div>
      );
    }
    return (
      <div className="comment-container">
        <div className="comment-author">
          <img src={author.image} alt="" />
        </div>

        <div className="comment-body">
          <Link className="author-name" to={`/profile/${author.username}`}>
            {author.username}
          </Link>
          <p className="comment-body">
            {body} <br />
            <Moment className="time-stamp" fromNow>
              {updatedAt}
            </Moment>
          </p>
          {user && (
            <div>
              {this.canEdit(
                user.username === author.username,
                user.role !== 'User',
              ) && (
                <FontAwesomeIcon
                  icon={faTrash}
                  className="delete"
                  onClick={() => deleteComment(slug, id)}
                  test-data="deleteButton"
                />
              )}
              {this.canEdit(
                user.username === author.username,
                user.role !== 'User',
              ) && (
                <FontAwesomeIcon
                  icon={faPencilAlt}
                  className="edit"
                  onClick={() => toggleEditCommentForm()}
                  test-data="editButton"
                />
              )}
              {history && history.length !== 0 && (
                <FontAwesomeIcon
                  icon={faHistory}
                  className="edit editHistory"
                  onClick={() => toggleEditHistory()}
                  test-data="trackEditButton"
                />
              )}
              <span className="buttonSpan">
                <button
                  type="button"
                  onClick={() => this.likeComment(id)}
                  id="like"
                  className="likeButton"
                >
                  <FontAwesomeIcon size="lg" icon={faThumbsUp} />
                </button>
              </span>
              <span>{likes}</span>
            </div>
          )}
          <div>
            <br />
          </div>
          <br />
          {editHistoryMode && history.length !== 0 && commentHistoryId === id && (
            <div
              className="edit-history-section"
              test-data="edit-history-section"
            >
              <FontAwesomeIcon
                icon={faEyeSlash}
                className="delete hideButton"
                onClick={() => toggleEditHistory(true)}
                test-data="closeButtonEditHistory"
              />
              {history.length > 0 &&
                history.map(oneHistory => {
                  return <EditHistory history={oneHistory} />;
                })}
            </div>
          )}
        </div>
      </div>
    );
  }
}
export const mapStateToprops = state => ({
  likecomment: state.like,
});
export default connect(
  mapStateToprops,
  { LikeComment },
)(Comment);
