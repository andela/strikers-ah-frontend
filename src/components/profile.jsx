import React, { Component } from 'react';
import { connect } from 'react-redux';
// import LeftProfile from './common/leftProfile';
import RightProfile from './common/profileRightSide';
import { getUserProfile, getUserArticles } from '../redux/actions/userAction';
import '../styles/css/profile.css';
import { getLoggedInUser } from '../helpers/authentication';

/**
 * @author Mwibutsa FLoribert
 * @returns { * } --
 */
class Profile extends Component {
  /**
   * @author Mwibutsa Floribert
   * @param {*} props
   */
  constructor(props) {
    super(props);
    this.state = {
      editProfile: { showForm: false, showImageForm: false },
      pagination: { pageSize: 5, currentPage: 0 },
    };
    const {
      getUserProfile: getProfile,
      getUserArticles: getArticles,
    } = this.props;
    const { username } = this.props;
    getProfile(username);
    getArticles(username);
  }

  /**
   * @author Mwibutsa Floribert
   * @param { * } nextPage --
   * @param { * } pages --
   * @param { * } prev --
   * @param { * } next --
   * @returns { * } --
   */
  paginate = (nextPage, pages, prev = false, next = false) => {
    const { pagination } = this.state;
    let activePage = pagination.currentPage;
    if (prev) {
      if (pagination.currentPage === 0) activePage = pages - 1;
      else activePage -= 1;
    } else if (next) {
      if (pagination.currentPage === pages - 1) activePage = 0;
      else activePage += 1;
    } else {
      activePage = nextPage;
    }
    this.setState({ pagination: { ...pagination, currentPage: activePage } });
  };

  toggleEditProfile = () => {
    const { editProfile } = this.state;
    this.setState({ editProfile: { showForm: !editProfile.showForm } });
  };

  toggleShowImage = () => {
    const { editProfile } = this.state;
    this.setState({
      editProfile: { showImageForm: !editProfile.showImageForm },
    });
  };
  /**
   * @author Mwibutsa FLoribert
   * @param { Integer } loggedInUserId --
   * @param { Integer } userProfileId --
   * @returns { Boolean } returns the comparison of logged in user id and user profile
   */

  isOwner = (loggedInUserId, userProfileId) => {
    return loggedInUserId === userProfileId;
  };

  /**
   * @author Mwibutsa FLoribert
   * @returns { JSX } --
   */
  render() {
    const { accountInfo: user } = this.props;
    const { articles } = this.props;
    let userArticles = [];
    if (Object.keys(articles).length) {
      userArticles = articles.articles;
    }
    let { profile } = user;
    profile = profile || user;
    const defaultUser = {
      name: 'Not Provided',
      image:
        'https://cdn.pixabay.com/photo/2013/07/13/10/44/man-157699__480.png',
      bio: '',
      followers: [],
      followings: [],
    };
    const isLoggedIn = !!getLoggedInUser();
    const owner = getLoggedInUser().id === profile.id;
    let ownerClass = 'profile-page';
    if (!isLoggedIn || !owner) {
      ownerClass += ' friend-profile';
    }
    const { editProfile, pagination } = this.state;
    return (
      <div className={ownerClass} test-data="profile-page" id="profile-page">
        <div className="profile-grid">
          {profile.email && (
            <RightProfile
              accountInfo={{
                ...defaultUser,
                ...profile,
                owner,
              }}
              editProfile={{
                showForm: editProfile.showForm,
                toggleEditProfile: this.toggleEditProfile,
                toggleShowImage: {
                  visible: editProfile.showImageForm,
                  toggleForm: this.toggleShowImage,
                },
              }}
              userArticles={userArticles}
              pagination={pagination}
              paginate={this.paginate}
            />
          )}
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => {
  return {
    accountInfo: state.userProfile.user,
    articles: state.userArticles,
  };
};
export default connect(
  mapStateToProps,
  { getUserProfile, getUserArticles },
)(Profile);
