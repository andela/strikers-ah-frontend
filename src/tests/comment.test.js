import React from 'react';
import JsonWebToken from 'jsonwebtoken';
import { shallow } from 'enzyme';
import '../enzymeConfig';
import { getComponent, findByTestAttribute } from './profile/test-helpers';
import { Comment, mapStateToprops } from '../components/comment/comment';

describe('TEST COMMENT', () => {
  let wrapper;
  let props;
  let token;
  let commentComponent;
  beforeEach(() => {
    token = JsonWebToken.sign({ id: 1 }, 'secretKey');
    localStorage.setItem('token', token);
    props = {
      comment: {
        comment: { body: 'this body', id: 1 },
        author: { image: 'image.png' },
      },
      deleteComment: jest.fn(),
      slug: 'Slug',
      editComment: jest.fn(),
      formId: 1,
      toggleEditCommentForm: jest.fn(),
      editMode: true,
      LikeComment: jest.fn(),
      likecomment: {
        message: 'comment liked',
      },
    };
    wrapper = getComponent(<Comment {...props} />);
    commentComponent = wrapper.instance();
  });
  it('Should render a comment component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should render a comment component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should be able to delete a comment', () => {
    const button = findByTestAttribute(wrapper, 'deleteButton');
    button.simulate('click', {});
    expect(commentComponent.props.deleteComment).toHaveBeenCalled();
  });
  it('should be able to edit  a comment', () => {
    const button = findByTestAttribute(wrapper, 'editButton');
    button.simulate('click', {});
    expect(commentComponent.props.toggleEditCommentForm).toHaveBeenCalled();
  });
  test('should test the like button', () => {
    const Wrapper = shallow(<Comment {...props} />);
    const spy = jest.spyOn(Wrapper.instance(), 'likeComment');
    Wrapper.update();
    const likeButton = Wrapper.find('#like');
    likeButton.simulate('click');
    expect(spy).toHaveBeenCalled();
  });
  test('should show the last displayed value', () => {
    const mockedState = {
      like: {
        like: {
          likecomment: 'comment liked',
        },
      },
    };
    const state = mapStateToprops(mockedState);
    expect(state).toBeTruthy();
  });
});
