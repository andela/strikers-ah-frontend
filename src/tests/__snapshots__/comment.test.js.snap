// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST COMMENT Should render a comment component 1`] = `ShallowWrapper {}`;

exports[`TEST COMMENT Should render a comment component 2`] = `ShallowWrapper {}`;
